src/partition.rs

```diff
@@ -563,16 +563,9 @@ pub fn get_intra_edges<T: Pixel>(
       needs_left = (!dc_or_cfl || x != 0)
         && !(mode == PredictionMode::D45_PRED
           || mode == PredictionMode::D67_PRED);
-      needs_topleft = mode == PredictionMode::V_PRED
-        || mode == PredictionMode::H_PRED
-        || mode == PredictionMode::D45_PRED
-        || mode == PredictionMode::D135_PRED
-        || mode == PredictionMode::D113_PRED
-        || mode == PredictionMode::D157_PRED
-        || mode == PredictionMode::D203_PRED
-        || mode == PredictionMode::D67_PRED
-        || mode == PredictionMode::PAETH_PRED;
-      needs_top = !dc_or_cfl || y != 0;
+      needs_topleft =
```

Comment:
This is good to simplify now, but not all directional modes need the top-left pixel if IEF is disabled. Since I am planning to fix the top-left pixel pre-processing for IEF and put it in this function it will probably end up being reverted.

Reply:
OK. I understand this simplification need to be changed after fixing #2107.
It is better to keep it as now.

Reply:
This part is changed to original and change for "needs_top" is kept.